plugins/postgres/dbt/include/postgres/macros/adapters.sql

```diff
@@ -71,6 +71,14 @@
 
 {% macro postgres__list_relations_without_caching(schema_relation) %}
   {% call statement('list_relations_without_caching', fetch_result=True) -%}
+    select
+      '{{ schema_relation.database }}' as database,
+      foreign_table_name as name,
+      foreign_table_schema as schema,
+      'table' as type
+    from information_schema.foreign_tables
+    where foreign_table_schema ilike '{{ schema_relation.schema }}'
```

Comment:
@lialzm could you take a look at the prototype version of this macro [here](https://github.com/dbt-labs/dbt-core/issues/7663#issuecomment-1584854886)?

### Key thing
The key thing is that it uses `external` as the type rather than `table`. I don't know for certain, but I'm guessing that we'll want to to be `external`.

i.e., we want it to be the same as `dbt_utils.get_table_types_sql` [here](https://github.com/dbt-labs/dbt-utils/blob/6ba7b660b1d1b4e3b41cb0cf6c3c0e4a70ae54e4/macros/sql/get_table_types_sql.sql#L18).

### Something else to consider

I don't have strong opinions on using the `information_schema.foreign_tables` view versus the `information_schema.tables` view. Using `information_schema.foreign_tables` is a little more simple since it doesn't need an extra filter in the where clause.